### PR TITLE
Initialize nouroutine with multiple modules

### PR DESCRIPTION
**Pull request by Ranko95**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [x] description of changes is added in CHANGELOG.md
- [x] update .d.ts typings

### CODE REVIEW

**Comment by Ranko95** [comment]
@tshemsedinov

File: noroutine.js [hunk 1]
```diff
@@ -110,10 +112,12 @@ const init = (options) => {
       throw new Error(`Norutine.init: options.${key} should be integer`);
     }
   }
-  balancer.target = findModule(options.module);
-  wrapModule(options.module);
+  balancer.targets = options.modules.map((m) => findModule(m));
```
Comment (tshemsedinov) [inline noroutine.js:115]:
```suggestion
  balancer.targets = options.modules.map(findModule);
```

File: lib/worker.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 'use strict';
 
 const { parentPort, workerData } = require('worker_threads');
-const target = require(workerData.module);
+
+const resolvedModules = workerData.modules.map((m) => require(m));
```
Comment (tshemsedinov) [inline lib/worker.js:5]:
```suggestion
const resolvedModules = workerData.modules.map(require);
```

Comment (tshemsedinov) [reply lib/worker.js:5]: You can rewrite this with reduce in 1 line

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
Great work, just need a few fixes

File: README.md [hunk 3]
```diff
@@ -18,19 +18,21 @@ Install: `npm install noroutine`
 ```js
 const noroutine = require('noroutine');
 const module1 = require('./module1.js');
-noroutine.init({ module: module1 });
+const module2 = require('./module2.js);
+noroutine.init({ modules: [module1, module2] });
 
 (async () => {
   const res = await module1.method1('value');
```
Comment (tshemsedinov) [inline README.md:25]:
```suggestion
  const res1 = await module1.method1('value1');
```

File: README.md [hunk 4]
```diff
@@ -18,19 +18,21 @@ Install: `npm install noroutine`
...
-  console.log({ res });
+  const res1 = await module2.method2('value1');
+  console.log({ res, res1 });
```
Comment (tshemsedinov) [inline README.md:27]:
```suggestion
  const res2 = await module2.method2('value2');
  console.log({ res1, res2 });
```

File: lib/worker.js [hunk 5]
```diff
@@ -1,7 +1,11 @@
 'use strict';
 
 const { parentPort, workerData } = require('worker_threads');
-const target = require(workerData.module);
+
+const target = workerData.modules.reduce(
+  (acc, module) => ({ ...acc, ...require(module) }),
+  {}
+);
```
Comment (tshemsedinov) [inline lib/worker.js:8]:
```suggestion
const names = workerData.modules;
const target = names.reduce((o, name) => ({ ...o, ...require(name) }), {});
```

**Comment by tshemsedinov** [comment]
@Ranko95 see my commit, going to land this PR after that

**Comment by Ranko95** [comment]
@tshemsedinov , saw your commit, thanks for suggestions!

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/noroutine/commit/402f3ec14a3fa8aa69b6e04f30a66a15c4527e5f

**Pull request closed without merging**